cmd/kubeadm/app/util/version.go

```diff
@@ -72,46 +75,49 @@ func KubernetesReleaseVersion(version string) (string, error) {
 	}
 
 	// revalidate, if exact build from e.g. CI bucket requested.
-	ver = normalizedBuildVersion(versionLabel)
-	if len(ver) != 0 {
+	if ver, err := normalizeBuildVersion(versionLabel); err == nil {
 		return ver, nil
 	}
 
 	// kubeReleaseLabelRegex matches labels such as: latest, latest-1, latest-1.10
-	if kubeReleaseLabelRegex.MatchString(versionLabel) {
-		// Try to obtain a client version.
-		// pkgversion.Get().String() should always return a correct version added by the golang
-		// linker and the build system. The version can still be missing when doing unit tests
-		// on individual packages.
-		clientVersion, clientVersionErr := kubeadmVersion(pkgversion.Get().String())
-		// Fetch version from the internet.
-		url := fmt.Sprintf("%s/%s.txt", bucketURL, versionLabel)
-		body, err := fetchFromURL(url, getReleaseVersionTimeout)
-		if err != nil {
-			// If the network operaton was successful but the server did not reply with StatusOK
-			if body != "" {
-				return "", err
-			}
-			// Handle air-gapped environments by falling back to the client version.
-			klog.Infof("could not fetch a Kubernetes version from the internet: %v", err)
-			klog.Infof("falling back to the local client version: %s", clientVersion)
-			return KubernetesReleaseVersion(clientVersion)
-		}
+	if !kubeReleaseLabelRegex.MatchString(versionLabel) {
+		return "", errors.Errorf("version %q doesn't match patterns for neither semantic version nor labels (stable, latest, ...)", version)
+	}
 
-		if clientVersionErr != nil {
-			klog.Warningf("could not obtain client version; using remote version: %s", body)
-			return KubernetesReleaseVersion(body)
+	// Try to obtain a client version.
```

Comment:
I think this should be moved after resolving the version label, so it will read:
- if it is a version label, resolve it
- get the kubeadm version
- ....

Reply:
ok, let's leave as it is